linera-chain/src/certificate.rs

```diff
@@ -88,6 +91,23 @@ impl<'de> Deserialize<'de> for ValidatedBlockCertificate {
     }
 }
 
+impl Serialize for TimeoutCerificate {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let cert = Certificate::from(self.clone());
+        cert.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for TimeoutCerificate {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let cert = Certificate::deserialize(deserializer)?;
+        TimeoutCerificate::try_from(cert).map_err(<D::Error as serde::de::Error>::custom)
```

Comment:
Ok so this won't work because `serde_reflection::trace_type` must produce a value of type `Certificate` line 106 but it has no way to know which variant of `Certificate` will work line 107.

Reply:
Done. I've implemented the (de)serializers manually.